org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Scanner.java

```diff
@@ -1980,6 +1981,7 @@ protected int getNextToken0() throws InvalidInputException {
 											}
 										}
 										if (!lineBeginsWithMarkdown()) {
+											this.currentPosition--;
```

Comment:
Reverting this one line change brings back proper rendering to HTML and fixes org.eclipse.jdt.ui.tests.hover.MarkdownCommentTests

Found this only by experimentation, have no idea what I'm talking about :)